src/TextRasterizer_P.cpp

```diff
@@ -428,9 +433,71 @@ sk_sp<SkImage> OsmAnd::TextRasterizer_P::rasterize(
     return target.asImage();
 }
 
+void OsmAnd::TextRasterizer_P::drawText(SkCanvas& canvas,
+                                        const TextPaint& textPaint,
+                                        const SkFont& font,
+                                        const SkPaint& paint) const
+{
+#ifndef OSMAND_USE_HARFBUZZ
+    canvas.drawSimpleText(
+        textPaint.text.constData(), textPaint.text.length()*sizeof(QChar), SkTextEncoding::kUTF16,
+        textPaint.positionedBounds.left(), textPaint.positionedBounds.top(),
+        font, paint);
+#else
+    QString inpStr = textPaint.text.toString();
+    auto textQStr = ICU::convertToVisualOrder(inpStr);
+
+    auto hb_font = textPaint.faceData->hbFont.get();
+    hb_font_set_scale(hb_font,
```

Comment:
Btw, we shouldn't modify that reused hb_font it would seem. I haven't noticed this, yet effectively hb_font in Typeface should be const, it would seem (skTypeface as well?). Maybe creating font every time is actually justified because of that - or at least look into hb_font_create_sub_font

Reply:
Yes, agree.
Let's clarify some points:
Is there are any reason to update font in the runtime (update font size and etc.)? (my assumption that answer NO, so in case of it's correct assumption then I'll move it to the initialization, and it looks like [this code](https://github.com/osmandapp/OsmAnd-core-legacy/blob/ee3d2f6838b74f05a657df6262e3b91617e53c58/native/src/textdraw.cpp#L903) should be updated too)
As a solution there are we could do all of this preparation once, and implement some function like `::update(scale)` in case of font size will need to be updated in the runtime for instance.

Reply:
> Is there are any reason to update font in the runtime (update font size and etc.)? 

How do you propose to render text of different size otherwise?

> ::update(scale)

Update won't be available on const object, and should not be, since it's shared.

Reply:
`hb_font_create_sub_font()` - creates full clone of parent font, thus it's not reasonable to use it here.
Will leave it as is (store `font` as a part of `typeface`, and set scale every-time here), based on our discussion.

Reply:
You should not modify what's passed as const

Reply:
Implemented using `hb_font_create_sub_font(...)`